### PR TITLE
[12.0] [ADD] sale_timesheet_purchase: Backport the module from v13 to v12.

### PR DESCRIPTION
**Pull request by Chandresh-SerpentCS**

### CODE REVIEW

**Comment by Chandresh-SerpentCS** [comment]
@max3903 @dreispt Request for review.

**Comment by dreispt** [comment]
@Chandresh-SerpentCS Please make Mr Travis happy

File: sale_timesheet_purchase/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+{
+    "name": "Sales Timesheet Purchase",
+    "category": "Hidden",
+    "version": "12.0.1.0.0",
+    "author": "Odoo S.A., Odoo Community Association (OCA)",
+    "license": "LGPL-3",
+    "summary": "Bridge module between Sales Timesheet and Purchase",
+    "depends": ["sale_timesheet", "purchase"],
+    "installable": True,
+    "auto_install": False,
```
Comment (dreispt) [inline sale_timesheet_purchase/__manifest__.py]: A bridge module is better auto installed
```suggestion
    "auto_install": True,
```

File: sale_timesheet_purchase/readme/DESCRIPTION.rst [hunk 2]
```diff
@@ -0,0 +1,4 @@
+sale_timesheet_purchase:
+
+Bridge module between Sales Timesheet and Purchase.
+The module is already in odoo v13. So, We contributed the same module in OCA v12.
```
Comment (dreispt) [inline sale_timesheet_purchase/readme/DESCRIPTION.rst]:
```suggestion
Allows to access purchase orders from Project Overview.

This is a bridge module between Sales Timesheet and Purchase.
It is an Odoo 13 standard module. This is a backport to Odoo 12.
```

File: sale_timesheet_purchase/readme/CONTRIBUTORS.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+* Chandresh Thakkar <cthakkar@opensourceintegrators.com>
```
Comment (dreispt) [inline sale_timesheet_purchase/readme/CONTRIBUTORS.rst:1]:
```suggestion
* Chandresh Thakkar <cthakkar@opensourceintegrators.com>
* Daniel Reis <dreis@opensourceintegrators.com>
```

File: sale_timesheet_purchase/controllers/main.py [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+from odoo import _, http
+
+from odoo.addons.sale_timesheet.controllers.main import SaleTimesheetController
+from odoo.addons.web.controllers.main import clean_action
+
+from odoo.http import request
+
+
+class SaleTimesheetControllerNew(SaleTimesheetController):
+    def _plan_get_stat_button(self, projects):
```
Comment (dreispt) [inline sale_timesheet_purchase/controllers/main.py:14]: This is overriding/replacing the original method.
You need to call super, and then enrich the returned `stat_buttons`, to add the Purchase Orders and Vendor Bills.
Needs fixing.

**Review by dreispt — CHANGES_REQUESTED** [review]

File: sale_timesheet_purchase/controllers/main.py (hunk 4 above)
Comment (dreispt) [reply sale_timesheet_purchase/controllers/main.py:14]: Chandresh, still no super() call

**Review by dreispt — CHANGES_REQUESTED** [review]
The `_plan_get_stat_button()` should call super() and then extend the result.

**Comment by dreispt** [comment]
If this is still hot in your plate, please squash commits.

**Comment by Chandresh-SerpentCS** [comment]
> If this is still hot in your plate, please squash commits.

Ok Sure. I will do.

**Review by dreispt — APPROVED** [review]
Comments addressed, LGTM

**Review by marcelsavegnago — APPROVED** [review]
LGTM

File: sale_timesheet_purchase/__manifest__.py (hunk 1 above)
Comment (nhomar) [reply sale_timesheet_purchase/__manifest__.py]: Mn this is a problem, with just be present in other environment this module stick as a glue... bad bad Idea :-/

**Pull request merged**